### PR TITLE
Cyclejs quickstart

### PR DESCRIPTION
**Pull request by atomrc**
@chenkie here is my PR for the Auth0 + Cycle.js quickstart based on the https://github.com/atomrc/cyclejs-auth0 package. 

Thanks for you feedback :)

### CODE REVIEW

**Comment by auth0build** [comment]
Can one of the admins verify this patch?

**Comment by chenkie** [comment]
Thanks very much Thomas! We'll review this shortly :)

File: snippets/client-platforms/cyclejs/dependencies.md [hunk 1]
```diff
@@ -0,0 +1 @@
+    npm install cyclejs-auth0 cyclic-router
```
Comment (chenkie) [inline snippets/client-platforms/cyclejs/dependencies.md]: This should be wrapped in triple ticks with `bash` as the lang :)

File: articles/client-platforms/cyclejs/01-login.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+### 1. Install the cyclejs-auth0 package
+
+To install both `cyclejs-auth0` and the `cyclic-router` (needed to parse the token sent by Auth0), from the command line, run:
+
+${snippet(meta.snippets.dependencies)}
+
+### 2. Instantiate the driver and configure the Auth0Lock
+
+In your main application file, you can now setup the `auth0Driver` and feed it you `client-ID` and `domain`:
```
Comment (chenkie) [inline articles/client-platforms/cyclejs/01-login.md]: Can we rename `client-ID` to `clientID` for consistency with other docs?

File: snippets/client-platforms/cyclejs/setup.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import {createHistory} from "history";
+import {makeRouterDriver} from 'cyclic-router'
+import {makeAuth0Driver, protect} from "cyclejs-auth0";
+
+function main(sources) {
+    //your application's code
+}
+
+const drivers = {
+    auth0: makeAuth0Driver("client-ID", "domain"),
```
Comment (chenkie) [inline snippets/client-platforms/cyclejs/setup.md]: We should use the code that automatically generates the user's ID and domain here :) 

`auth0: makeAuth0Driver('<%= account.clientId %>', '<%= account.namespace %>')`

File: articles/client-platforms/cyclejs/01-login.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+${snippet(meta.snippets.setup)}
+
+### 3. Implement the login
+
+Now that everything is setted, you can activate authentication on some of your components. Activating authentication on a component is a simple as calling the `protect` function on that component.
```
Comment (chenkie) [inline articles/client-platforms/cyclejs/01-login.md]: `...everything is setted...` => `...everything is set...`
`...is a simple...` => `...is as simple...`

File: articles/client-platforms/cyclejs/01-login.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Now that everything is setted, you can activate authentication on some of your components. Activating authentication on a component is a simple as calling the `protect` function on that component.  
+Let's asume you have a `Todos` component and you want to ensure the user is logged in to see it.
+
+${snippet(meta.snippets.use)}
+
+Now if the user is not logged in when the component is instantiated, the Auth0 form will show up.
+
+### 4. Configuring the login form
+
+You may want to use configure the behavior of the Auth0 login form. To achieve that, you can use the `auth0ShowParams` options on the `protect` function:
```
Comment (chenkie) [inline articles/client-platforms/cyclejs/01-login.md]: `...want to use configure...` => `...want to configure...`

File: articles/client-platforms/cyclejs/01-login.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+}
+```
+
+All the available arguments for `lock.show` are supported, see [.show\(\[options, callback\]\)](/libraries/lock#-show-options-callback-).
+
+After authentication, the `protect` function will handle the token parsing and storing it to `localStorage`.
+
+### 4. Configure secure calls to your API
+
+To configure secure calls to the API you are creating<%= configuration.api ? ' on ' + configuration.api : '' %>, you need to decorate all your newly secured component's http calls with the [JWT token](/jwt) that has been stored in `localStorage`. To do that, you can set a decorator that will be called on each http request and where you can add the `Authorization` header.
```
Comment (chenkie) [inline articles/client-platforms/cyclejs/01-login.md]: `...http calls...` => `...HTTP calls...`
`...JWT token...` => `...JSON Web Token...`

File: articles/client-platforms/cyclejs/01-login.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
+---
+title: Login
```
Comment (chenkie) [inline articles/client-platforms/cyclejs/01-login.md:2]: It looks like the hyperlinks in the doc are still pointing to localhost :)

File: articles/client-platforms/cyclejs/01-login.md (hunk 2 above)
Comment (atomrc) [inline articles/client-platforms/cyclejs/01-login.md]: I actually found the `client-ID` in your doc here : https://github.com/auth0/docs/blob/master/articles/client-platforms/jquery/01-login.md#2-configure-the-auth0lock

File: articles/client-platforms/cyclejs/01-login.md (hunk 7 above)
Comment (atomrc) [inline articles/client-platforms/cyclejs/01-login.md:2]: Not sure I understand what is the problem here? To which link are you referring to?

File: snippets/client-platforms/cyclejs/setup.md (hunk 3 above)
Comment (atomrc) [inline snippets/client-platforms/cyclejs/setup.md]: oh nice :)

Comment (atomrc) [inline snippets/client-platforms/cyclejs/setup.md]: What happens if the reader is not logged?

**Comment by atomrc** [comment]
Thanks for your review :) BTW is there a way I can see the final rendering of the article (on my machine or on heroku)?

Also I don't really understand what is wrong with the hyperlinks. I don't see any links pointing to localhost in my article

**Comment by chenkie** [comment]
Ah, the hyperlink issue might just be an artifact of the Heroku preview I'm looking at. Here's the preview :) https://auth0-docs-content-pr-1864.herokuapp.com/docs/quickstart/spa/cyclejs

**Comment by atomrc** [comment]
Ok so I think I fixed everything now. Is there anything else?

**Comment by chenkie** [comment]
Thanks @atomrc :) We'll do our final review cc/ @mpaktiti

**Comment by atomrc** [comment]
Oh awesome. Thanks guys.

Just one thing though, the icon doesn't seem to display on that page https://auth0.com/docs/quickstart/spa
Have I missed something. I cannot seem to find where are stored the logos in the repo?

**Comment by chenkie** [comment]
We need to update our style guide with the icon. It'll be updated soon 😄 

Thanks again for this

**Pull request merged**